Show $ANTI and $PRO balance next to wallet connect button

kokoTTM opened the pull request:
This feature will make the user feel more connected.

![image](https://github.com/user-attachments/assets/443d1232-3445-4603-8a18-08a8960de79e)
![image](https://github.com/user-attachments/assets/f057a103-1270-4966-8eda-bf9dcff65a58)

Commit:
show anti and pro balance (desktop)

Commit:
showshow anti and pro balance (mobile)

kokoTTM commented:
the mainnet rpc doesn't work locally, tested using mine

axispx commented on src/components/TokenBalance.js:
```diff
@@ -0,0 +1,33 @@
+      setAntiBalance(antiBalanceResult)
+      setProBalance(proBalanceResult)
+    }
+    if (wallet.publicKey)
```
either braces here or having the code in a single line would be preferred to prevent any bugs in the future

```suggestion
    if (wallet.publicKey) checkBalance()
```

axispx commented on src/utils/solana.js:
```diff
@@ -6,10 +6,20 @@ const connection = new Connection(clusterApiUrl("mainnet-beta"));
   );
-  return accounts.value.length > 0 && accounts.value[0].account.data.parsed.info.tokenAmount.uiAmount > 0;
+  return accounts.value?.reduce((memo, item) => {
+    return memo + item?.account?.data.parsed.info.tokenAmount.uiAmount || 0
```
would suggest to wrap the check for uiAmount

```suggestion
    return memo + (item?.account?.data.parsed.info.tokenAmount.uiAmount || 0);
```

axispx reviewed (changes requested):
Hi @kokoTTM thank you for the PR, means a lot to get PR from the community.

Checking all accounts instead of just one can be a good improvement but letting user select the account would be better option in my opinion. I'll let @sshmatrix decide on that one.

Meanwhile, I have requested some minor changes in the code, otherwise the code looks good to me.

 I'd also like to ask you to look at other approaches to display the price so that it blends well with the whole UI.

Thanks.

Commit:
review update

kokoTTM commented:
> Hi @kokoTTM thank you for the PR, means a lot to get PR from the community.
> 
> Checking all accounts instead of just one can be a good improvement but letting user select the account would be better option in my opinion. I'll let @sshmatrix decide on that one.
> 
> Meanwhile, I have requested some minor changes in the code, otherwise the code looks good to me.
> 
> I'd also like to ask you to look at other approaches to display the price so that it blends well with the whole UI.
> 
> Thanks.

Thanks for suggestions @axispx , updated accordingly.
for the 'price' UI, you mean the token balance I added right? if so ya would be helpful if we can have some design to refer to

sshmatrix commented:
> the mainnet rpc doesn't work locally, tested using mine

Is there an alternative to the solana.com RPC that works locally?

kokoTTM commented:
> > the mainnet rpc doesn't work locally, tested using mine
> 
> Is there an alternative to the solana.com RPC that works locally?
> 
> > for the 'price' UI, you mean the token balance I added right? if so ya would be helpful if we can have some design to refer to
> 
> What you see is the design we have :) Feel free to experiment and find your preferred optimal way to show the balance. Good to add the ANTI:PRO ratio in there as well!

for local development I use quicknode

Commit:
updated UI alignment

kokoTTM commented:
added ratio (by percentage)
adjusted desktop alignment, made it a separate line below:

![image](https://github.com/user-attachments/assets/6b5715d4-ef73-4803-b260-9cf583aaaf99)
![image](https://github.com/user-attachments/assets/1b6957ee-cc0f-4d77-ba30-6aa41cc4b85b)
